config_parser.cc

```diff
@@ -211,10 +210,13 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
         // Error.
         break;
       }
-      if(config_stack.size() == 1)
+      //Check that the config_stack does not have leftover tokens after parsing
+      //e.g. a config that has a '{' without the matching '}'
+      if(config_stack.size() == 1){
```

Comment:
The if should look like 

if (config_stack.size() == 1) {
  return true;
} else {
  break;
}